firmware/signing.c

```diff
@@ -990,12 +1016,40 @@ void signing_txack(TransactionType *tx)
 			return;
 
 		case STAGE_REQUEST_SEGWIT_INPUT:
+			progress = 500 + ((signatures * progress_step) >> PROGRESS_PRECISION);
 			resp.has_serialized = true;
 			resp.serialized.has_signature_index = false;
 			resp.serialized.has_signature = false;
 			resp.serialized.has_serialized_tx = true;
-			if (tx->inputs[0].script_type == InputScriptType_SPENDP2SHWITNESS
-				&& !tx->inputs[0].has_multisig) {
+			if (tx->inputs[0].script_type == InputScriptType_SPENDADDRESS) {
```

Comment:
Here  is the new bcc case.  It's  input are treated like segwit inputs, so that they don't require to serialize the whole  transaction.